src/app/modules/main/profile_section/notifications/module.nim

```diff
@@ -63,23 +106,72 @@ method viewDidLoad*(self: Module) =
 method getModuleAsVariant*(self: Module): QVariant =
   return self.viewVariant
 
-method unmuteChat*(self: Module, chatId: string) =
-  self.controller.unmuteChat(chatId)
-
-method onChatMuted*(self: Module, chatId: string) =
-  let chat = self.controller.getChatDetails(chatId)
-  if(chat.chatType == ChatType.OneToOne):
-    let (chatName, chatImage) = self.controller.getOneToOneChatNameAndImage(chat.id)
-    let item = initItem(chat.id, chatName, chatImage, chat.color)
-    self.view.mutedContactsModel().addItem(item)
+method sendTestNotification*(self: Module, title: string, message: string) =
+  singletonInstance.globalEvents.showTestNotification(title, message)
+
+method saveExemptions*(self: Module, itemId: string, muteAllMessages: bool, personalMentions: string, 
+  globalMentions: string, otherMessages: string) =
+  var allExemptions = singletonInstance.localAccountSensitiveSettings.getNotifSettingExemptionsAsJson()
+  allExemptions[itemId] = %* {
+    EXEMPTION_KEY_MUTE_ALL_MESSAGES: muteAllMessages,
+    EXEMPTION_KEY_PERSONAL_MENTIONS: personalMentions, 
+    EXEMPTION_KEY_GLOBAL_MENTIONS: globalMentions,
+    EXEMPTION_KEY_OTHER_MESSAGES: otherMessages
+  }
+  
+  self.view.exemptionsModel().updateExemptions(itemId, muteAllMessages, personalMentions, globalMentions, otherMessages)
+  
+  singletonInstance.localAccountSensitiveSettings.setNotifSettingExemptions($allExemptions)
+
+method onToggleSection*(self: Module, sectionType: SectionType) =
+  if(sectionType != SectionType.Community):
+    return
+
+  if(singletonInstance.localAccountSensitiveSettings.getCommunitiesEnabled()):
```

Comment:
@jrainville I don't see how? Add pseudo code here what's your idea?

Here is an explanation of the current code:
Code here should be applied only for the sections which are type of `SectionType.Community`. So idea is if we receive here anything but the `SectionType.Community` to get back immediately, otherwise we're taking action based on `getCommunitiesEnabled()` and reach `then` or `else` block depends on that condition.

Now if I merge `getCommunitiesEnabled()` to the condition in the line 127 either with `AND` or `OR` that won't be the same condition any more, cause `then` and `else` blocks won't be executed in the same cases as they are executed now.

Reply:
My bad, I missed the `else`